plugins/skipmap/index.js

```diff
@@ -0,0 +1,158 @@
+import { CHAT_MESSAGE, NEW_GAME } from 'squad-server/events';
+
+export default {
+  name: 'skipmap',
+  description:
+    'The `skipmap` plugin will allow players to vote via `+/-` if they wish to skip the map',
+
+  defaultEnabled: true,
+  optionsSpec: {
+    command: {
+      required: true,
+      description: 'The name of the command to be used in chat.',
+      default: '!skipmap'
+    },
+
+    voteDuration: {
+      required: false,
+      description: 'How long the vote should go.',
+      default: 5 * 60 * 1000
+    },
+
+    startTimer: {
+      required: false,
+      description: 'Time to wait before allowing the vote to go through.',
+      default: 15 * 60 * 1000
+    },
+
+    pastVoteTimer: {
+      required: false,
+      description: 'Time that needs to have passed since the last vote.',
+      default: 10 * 60 * 1000
+    },
+
+    endTimer: {
+      required: false,
+      description: 'Time before the end timer of the round where the votes are no longer valid.',
+      default: 15 * 60 * 1000
+    },
+
+    minimumVotes: {
+      required: false,
+      description: 'The minimum required amount of votes for the vote to go through',
+      default: 20
+    },
+
+    reminderInterval: {
+      required: false,
+      description: 'The time between individual reminders.',
+      default: 2 * 60 * 1000
+    }
+  },
+
+  init: async (server, options) => {
+    let votePos = 0;
+    let voteNeg = 0;
+    let intervalReminderBroadcasts;
+    let timeoutVote;
+    let timeLastVote;
+    let voteActive;
+    const playerVotes = {};
+
+    server.on(CHAT_MESSAGE, (info) => {
+      // Run through conditions
+      // check if message is command
+      if (!info.message.chat.startsWith(options.command)) return;
+      // check if enough time has passed since start of round and if not, inform the player
+      if (server.layerHistory[0].time < options.startTimer) {
+        const min = Math.floor(
+          ((options.startTimer - server.layerHistory[0].time) % (1000 * 60 * 60)) / (1000 * 60)
+        );
+        const sec = Math.floor(
+          (options.startTimer - server.layerHistory[0].time) % ((1000 * 60) / 1000)
+        );
+        server.rcon.warn(
+          info.steamID,
+          'Not enough time has passed since the start of the match. Please try again in ' +
+            min +
+            'm ' +
+            sec +
+            's'
+        );
+        return;
+      }
+      // check if enough time remains in the round, if not, inform player
+      if (server.layerHistory[0].time < 2 * 60 * 60 * 1000 - options.endTimer) {
```

Comment:
I'd probably rework this condition a little as matches rarely last 2 hours. Instead, I'd maybe change this condition to "is it within x amount of the start time?", which would allow servers to go you can skip map providing it's between 5 and 10 minutes of the match starting using a combination of conditions. This might be more appealing as some servers may wish to block maps being changed after teams have already sunken 30 minutes of effort into the round.